virtualmachine/gcp/vm.go

```diff
@@ -206,10 +335,366 @@ func (vm *VM) DeleteDisks() error {
 	return nil
 }
 
-func (vm *VM) AddDisk() error {
-        return errors.New("TO-DO")
+// This function cannot be used in GCP as defined in the interface. The
+// function's signature will be different for vsphere, gcp and aws. This can be
+// cleaned up when the VirtualMachine interface is corrected. Please refer to
+// AddNewDisks function to add new disks to an instance.
+func (vm *VM) AddDisk() ([]string, error) {
```

Comment:
This interface has been getting corrected through other PR. You can either remove or at least add a todo-comment to remove this code.